src/openforms/submissions/tests/factories.py

```diff
@@ -326,7 +333,12 @@ def content_type(self) -> str:
 
 class SubmissionFileAttachmentFactory(factory.django.DjangoModelFactory):
     submission_step = factory.SubFactory(SubmissionStepFactory)
-    temporary_file = factory.SubFactory(TemporaryFileUploadFactory)
+    temporary_file = factory.SubFactory(
+        TemporaryFileUploadFactory,
+        submission=factory.LazyAttribute(
+            lambda o: o.factory_parent.submission_step.submission
+        ),
+    )
```

Comment:
is `factory.SelfAttribute` not an option? I think `factory_parent` is private API

Reply:
They both work and `factory_parent` is documented as an alternative: https://factoryboy.readthedocs.io/en/stable/reference.html#parents